ansible/roles/aws_service_limit/library/aws_service_limit.py

```diff
@@ -0,0 +1,187 @@
+
+    Param: ec2 connection object
+    Returns: list of account attributes; each attribute is a dict
+    '''
+
+    # Gather results from EC2
+    attr_results = ec2.describe_account_attributes()
+
+    # strip out extraneous info
+    attributes = []
+    for attr in attr_results['AccountAttributes']:
+        if len(attr['AttributeValues']) < 2:
+            attributes.append({attr['AttributeName']:attr['AttributeValues'][0]['AttributeValue']})
+        else:
+            vals = []
+            for val in attr['AttributeValues']:
+                vals.append(val['AttributeValue'])
+            attributes.append({attr['AttributeName'] : vals})
+    return attributes
+
+def service_limits(tr_adv):
+    '''
+    Collects and arranges Trusted Advisor service limits
+
+    Param: support connection object
+    Returns: list of service limits; each service limit is a dict containing:
+        - region (string)
+        - service (string)
+        - limit_name (string)
+        - limit_value (integer)
+        - current_value (integer)
+        - status (string)
+    '''
+    # look up the service limits ID
+    ta_query = tr_adv.describe_trusted_advisor_checks(language='en')
+    service_limits_id = None
+    for check in ta_query['checks']:
+        if check['name'] == "Service Limits":
+            service_limits_id = check['id']
+
+    # Gather results from Trusted Advisor
+    ta_results = tr_adv.describe_trusted_advisor_check_result(checkId=service_limits_id)
+
+    # extract limits from the most recent TA report
+    limits = []
+    for res in ta_results['result']['flaggedResources']:
+        limits.append({'region' : res['metadata'][0],
+                       'service' : res['metadata'][1],
+                       'limit_name' : res['metadata'][2],
+                       'limit_value' : res['metadata'][3],
+                       'current_value' : res['metadata'][4],
+                       'status' : res['metadata'][5]})
+    return limits
+
+def main():
+    """
+    get service limits
+    :return:
+    """
+
+    result = {}
+
+    # Including ec2 argument spec
+    module = AnsibleModule(argument_spec=ec2_argument_spec(),
+                           supports_check_mode=True)
+
+    # Set up connection
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module,
+                                                                  boto3=HAS_BOTO3)
+
+    # Set up connection
+    if region:
+        try:
+            ec2_conn = boto3_conn(module,
+                                  conn_type='client',
+                                  resource='ec2',
+                                  region=region,
+                                  endpoint=ec2_url,
+                                  **aws_connect_params)
+
+            ta_conn = boto3_conn(module,
+                                 conn_type='client',
+                                 resource='support',
+                                 region=region,
```

Comment:
Found the answer. There's only one support endpoint. (See: https://docs.aws.amazon.com/general/latest/gr/rande.html#awssupport_region )

I'll fix this.